enhancements/sig-architecture/70-spread-policy/README.md

```diff
@@ -0,0 +1,288 @@
+
+**Example.** We want six clusters to deploy our backend service.
+Moreover, we want 2/3 of them to be located in us-east-1 and 1/3 to be located in us-west-1 since there are more clients from the east coast of the US.
+
+## Goal
+Generally, we have two goals for the spread policy:
+1. Cover the full functionality in the Motivation section;
+2. Implement a high-performance and scalable mechanism that works for thousands of clusters efficiently.
+
+## Proposal
+
+### API Spec
+```go
+type PlacementSpec struct {
+	// ...
+
+	// SpreadPolicy defines how placement decisions should be distributed among a
+	// set of clusters.
+	// +optional
+	SpreadPolicy SpreadPolicy `json:"spreadPolicy,omitempty"`
+}
+
+// SpreadPolicy defines how placement decisions should be spreaded among the
+// clusters.
+type SpreadPolicy struct {
+	// SpreadConstraints defines how placement decision should be distributed among a
+	// set of clusters.
+	// +optional
+	// +kubebuilder:validation:MaxItems=64
+	SpreadConstraints []SpreadConstraintsTerm `json:"spreadConstraints,omitempty"`
+}
+
+// SpreadConstraintsTerm defines a terminology to spread placement decisions
+type SpreadConstraintsTerm struct {
+	// TopologyKey is either a label key or a cluster claim name of ManagedClusters
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]$`
+	// +kubebuilder:validation:MaxLength=316
+	TopologyKey string `json:"topologyKey"`
+
+	// TopologyKeyType indicates the type of TopologyKey. It could be Label or Claim.
+	// +required
+	// +kubebuilder:validation:Required
+	TopologyKeyType TopologyKeyType `json:"topologyKeyType"`
+
+	// MaxSkew describes the degree to which the workload may be unevenly distributed.
+	// If not omitted, the scheduler would not schedule if MaxSkew cannot be satisfied.
+	// The minimum possible value is 1.
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	MaxSkew *int32 `json:"maxSkew,omitempty"`
+
+	// Weight indicates the weight of the SpreadPolicyTerm. It is required only when multiple
+	// terms are used. To spread the workload, the scheduler takes the even constraints with
+	// higher weight into prior consideration.
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:default=1
+	Weight int32 `json:"weight,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=Label;Claim
+type TopologyKeyType string
+
+const (
+	// Valid TopologyKeyType value is Claim, Label.
+	TopologyKeyTypeClaim TopologyKeyType = "Claim"
+	TopologyKeyTypeLabel TopologyKeyType = "Label"
+)
+```
+
+### API Example
+
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: Placement
+metadata:
+  name: demo-placement
+spec:
+  # numberOfClusters is required if spreadConstraints contains
+  # a constraint of Even type.
+  numberOfClusters: 4
```

Comment:
is number of clusters logically required to satisfy the skew?  I don't think it is, but it makes picking the values to put there slightly more difficult.

Reply:
If numberOfClusters is not specified, the behavior of the scheduler may be trying to select all clusters which meet the placement requirements, and check the skew. I agree that it could be optional.